backend/pkg/api/data_access/mobile.go

```diff
@@ -408,20 +398,20 @@ func (d *DataAccessService) GetValidatorDashboardMobileValidators(ctx context.Co
 			penalty_count,
 			node_deposit_balance,
 			status,
-			rn.smoothing_pool_opted_in
+			rn.smoothing_pool_opted_in,
+			validator_index
 		FROM rocketpool_minipools
 		LEFT JOIN rocketpool_nodes rn ON rocketpool_minipools.node_address = rn.address
-		WHERE pubkey = ANY($1)
+		WHERE validator_index = ANY($1)
 		`
-		err := d.alloyReader.SelectContext(ctx, &rocketPoolResults, validatorsQuery, pq.ByteaArray(pubKeys))
+		err := d.alloyReader.SelectContext(ctx, &rocketPoolResults, validatorsQuery, indices)
```

Comment:
Good point, they way I handled it is that it will use the result table of GetValidatorDashboardValidators to get the indices, not the full list of all indices. Basically this is already a truncated version of the indices involved and should never be more than whatever max rows `GetValidatorDashboardValidators` returns

Reply:
Ah right it's limited because you're passing the cursor/limit along, should be minimal impact then :+1: